open-policy-agent/trusted-image-sources/disallowed-registries/template.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sdisallowedregistries
+  annotations:
+    description: Requires setting up allowed image sources (registries). Any other image source is disallowed.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sDisallowedRegistries
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package K8sDisallowedRegisitries
+
+        missing(obj, field) = true {
+            not obj[field]
+        }
+
+        missing(obj, field) = true {
+            obj[field] == ""
+        }
+
+        violation[{"msg": msg}] {
+          input.review.object.kind == "Image"
+          input.review.object.apiVersion == "config.openshift.io/v1"
+          missing(input.review.object.spec, "registrySources")
+          msg := sprintf("%v object must have spec.registrySources.allowedRegistries configured", [input.review.object.kind])
+        }
+
+        violation[{"msg": msg}] {
+          input.review.object.kind == "Image"
+          input.review.object.apiVersion == "config.openshift.io/v1"
+          missing(input.review.object.spec.registrySources, "allowedRegistries")
+          msg := sprintf("%v object must have spec.registrySources.allowedRegistries configured", [input.review.object.kind])
+        }
+
+        violation[{"msg": msg}] {
+          input.review.object.kind == "Image"
+          input.review.object.apiVersion == "config.openshift.io/v1"
+          count(input.review.object.spec.registrySources.allowedRegistries) == 0
+          msg := sprintf("%v object must have at least one registry configured at spec.registrySources.allowedRegistries", [input.review.object.kind])
+        }
+
+        violation[{"msg": msg}] {
+          input.review.object.kind == "Image"
+          input.review.object.apiVersion == "config.openshift.io/v1"
+          allowedRegistries := { registry | registry := input.review.object.parameters.allowedRegistries[_]}
```

Comment:
Out of ignorance, why can you pass this in two different parameters to begin with?